File: src/main/java/esport/sprint1/sprint1/models/Console.java
package esport.sprint1.sprint1.models;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
public class Console implements Serializable{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Long id;

    private String nom;

    @ManyToMany(mappedBy = "consoles", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Jeu> jeux;

    @OneToMany(mappedBy = "console")
    private List<Tournois> tournois;

    public Console() {
    }

    public Console(String nom) {
        this.nom = nom;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public List<Jeu> getJeux() {
        return jeux;
    }

    public void setJeux(List<Jeu> jeux) {
        this.jeux = jeux;
    }

    public List<Tournois> getTournois() {
        return tournois;
    }

    public void setTournois(List<Tournois> tournois) {
        this.tournois = tournois;
    }
}

File: src/main/frontend/src/models/Console.ts
import { Jeu } from './Jeu';

export class Console {
  private _id: number;
  private _Nom: string;
  private _Jeux: Array<Jeu>;

  public constructor(
    _id: number,
    Nom: string,
    Jeux: Array<Jeu>) {
    this._id = _id;
    this._Nom = Nom;
    this._Jeux = Jeux;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get Nom(): string {
    return this._Nom;
  }

  set Nom(value: string) {
    this._Nom = value;
  }

  get Jeux(): Array<Jeu> {
    return this._Jeux;
  }

  set Jeux(value: Array<Jeu>) {
    this._Jeux = value;
  }
}
